src/components/Chatbox/Chatbox.tsx

```diff
@@ -42,15 +43,15 @@ type Props = {
   sendMessageBoxId?: string;
   messages?: ChatMessageList;
   isLoading?: boolean;
-  sendMessageFunction?: (message: PartialNewChatMessage) => void;
-  deleteMessageFunction?: (message: PartialChatMessage) => void;
-  editMessageFunction?: (message: PartialChatMessage) => void;
+  sendMessageFunction?: SendMessageFunctionType;
+  deleteMessageFunction?: DeleteMessageFunctionType;
```

Comment:
And here I don't know if it would be possible to do `typeof useDeleteMessage`?

Reply:
Yes it would be possible and better to derive the type of the function from the hook itself !